### PR TITLE
[chore] Prep for v3.0.0 release

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Merge in all changes for v3
- Bump version number

# Testing

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc.)

### CODE REVIEW

**Comment by nwithan8** [comment]
This PR will be "merged", not "squashed", to preserve history.

**Comment by jchen293** [comment]
You need to rebase this

**Comment by nwithan8** [comment]
> You need to rebase this

No, this is the branch that has been collecting all the breaking changes for v3. It's not a rebase error.

**Comment by jchen293** [comment]
> > You need to rebase this
> 
> No, this is the branch that has been collecting all the breaking changes for v3. It's not a rebase error.

Ah yes. Didn't see you are merging into master. I will stamp once the build is green

File: Makefile [hunk 1]
```diff
@@ -43,7 +43,7 @@ scan:
 
 ## test - Test the project
 test:
-	go clean -testcache && go test ./tests -v
+	go clean -testcache && go test ./tests -v -mod=mod
```
[the remark this answers is not in the document]
Comment (nwithan8) [reply Makefile]: Resolved

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go-version: ['1.15', '1.16', '1.17', '1.18', '1.19', '1.20']
+        go-version: ['1.16', '1.17', '1.18', '1.19', '1.20']
```
Comment (Justintime50) [inline .github/workflows/ci.yml:30]: thought: wanna add `1.21` to the end of the list? It just came out yesterday

**Pull request merged**